### PR TITLE
Add Occupations controller

### PR DESCRIPTION
**Pull request by jenlu33**
Asana ticket: [https://app.asana.com/0/1203289004376659/1205503615126504/f](https://app.asana.com/0/1203289004376659/1205503615126504/f)

This adds an Occupations controller in preparation for using occupations in the typeahead results.

### CODE REVIEW

File: app/controllers/occupations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class OccupationsController < ApplicationController
+  def index
+    es_response = OccupationElasticsearchQuery.new(
+      search_params: params.permit(:q).to_h
+    ).call
+    occupations = es_response.records.to_a
```
Comment (FerPerales) [inline app/controllers/occupations_controller.rb:6]: *nit*: Ypu may not need to call `to_a`. I removed it for occupation standards index and is working 😄

Comment (jenlu33) [reply app/controllers/occupations_controller.rb:6]: The tests were failing when I tried to to remove `to_a` from occupations

File: app/models/occupation.rb [hunk 2]
```diff
@@ -14,22 +14,50 @@ class Occupation < ApplicationRecord
...
+      indexes :title, type: :text, analyzer: :autocomplete, search_analyzer: :autocomplete_search
+      indexes :rapids_code, type: :text, analyzer: :autocomplete, search_analyzer: :autocomplete_search
+      indexes :onet_code, type: :text, analyzer: :autocomplete, search_analyzer: :autocomplete_search
     end
   end
 
   def as_indexed_json(_ = {})
     as_json(
-      only: [:title]
+      only: [:title, :rapids_code, :onet_code]
```
Comment (littleforest) [inline app/models/occupation.rb:51]:
```suggestion
      only: [:title, :rapids_code]
```
This top section is only for fields on the model.

File: spec/queries/occupation_elasticsearch_query_spec.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    onet3 = create(:onet, code: "11.2345")
+
+    o1 = create(:occupation, onet: onet1)
+    create(:occupation, onet: onet2)
+    create(:occupation, onet: onet3)
+
+    Occupation.import
+    Occupation.__elasticsearch__.refresh_index!
+
+    params = {q: "12.3456"}
```
Comment (littleforest) [inline spec/queries/occupation_elasticsearch_query_spec.rb:54]: What about making this one have a search term of "12.345" and showing that it returns the first and second occupations?

File: spec/requests/occupations_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+require "rails_helper"
+
+RSpec.describe "Occupation", type: :request do
+  describe "GET /index.json", :elasticsearch do
+    it "returns a json response" do
+      create(:occupation, title: "Mechanic")
```
Comment (littleforest) [inline spec/requests/occupations_spec.rb:6]: Let's test all search results here:

```
onet = create(:onet, code: "12-3456")
create(:occupation, title: "Mechanic", rapids_code: "5678CB", onet: onet)
```

(see other additions below)

File: spec/requests/occupations_spec.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      Occupation.import
+      Occupation.__elasticsearch__.refresh_index!
+
+      get occupations_path, params: {format: "json", q: "Mech"}
+
+      expect(response_json[0][:display]).to eq "Mechanic"
+      expect(response_json[0][:link]).to eq "/occupations"
+
+      expect(response).to be_successful
+      expect(response.content_type).to eq "application/json; charset=utf-8"
```
Comment (littleforest) [inline spec/requests/occupations_spec.rb:17]: And then add additional checks for matching on onet_code and rapids_code:

```
      get occupations_path, params: {format: "json", q: "12-34"}

      expect(response_json[0][:display]).to eq "Mechanic"
      expect(response_json[0][:link]).to eq "/occupations"

      get occupations_path, params: {format: "json", q: "567"}

      expect(response_json[0][:display]).to eq "Mechanic"
      expect(response_json[0][:link]).to eq "/occupations"
```

**Review by littleforest — COMMENTED** [review]
Nice!

**Review by littleforest — APPROVED** [review]
🚢

**Pull request merged**